### PR TITLE
grpc streaming of events

### PR DESCRIPTION
**Pull request by mkundu1**

### CODE REVIEW

File: protobufs/settings_engine.proto [hunk 1]
```diff
@@ -52,7 +52,7 @@ message SetStateEvent {
   map<string, double> state = 1;
 }
 
-message EventReply {
+message Event {
```
Comment (seanpearsonuk) [inline protobufs/settings_engine.proto:55]: @mkundu1 over all, the streamed response messages need to encapsulate the state + metadata (constraints: allowed_values, min, max, and potentially also is_read_only)?  Those are all the data that need to go back to the subscriber in general. 

Some subscribers might not wish to hear about constraint changes (e.g., it's just assembling the current state in order to execute on it or just display (without allowing modification) or save it). So, shall we reflect that in the subscribe action?

Shall we separate out the metadata from the state data in these streamed responses? That seems like a clean approach, especially given that only some subscribers will require the metadata.

**Comment by seanpearsonuk** [comment]
@mkundu1 please could you Krishnan to all the repos and thereby to the PRs too?  Thanks.

File: protobufs/settings_engine.proto (hunk 1 above)
Comment (mkundu1) [reply protobufs/settings_engine.proto:41]: Shall we have two response-streaming rpcs? I think having single rpc streaming the full effect is better as client can process all of it together.

Comment (mkundu1) [reply protobufs/settings_engine.proto:50]: Added these information in the proto file (not implemented yet)

**Pull request merged**